packages/cfd/src/Containers/__tests__/compare-accounts-content.spec.js

```diff
@@ -0,0 +1,202 @@
+
+describe('should render the CompareAccountsContent component properly', () => {
+    let modal_root_el;
+
+    beforeAll(() => {
+        modal_root_el = document.createElement('div');
+        modal_root_el.setAttribute('id', 'modal_root');
+        document.body.appendChild(modal_root_el);
+    });
+
+    afterAll(() => {
+        document.body.removeChild(modal_root_el);
+    });
+
+    const all_landing_companies = {
+        mt_gaming_company: {
+            financial: 'USD',
+        },
+        mt_financial_company: {
+            financial: 'USD',
+            financial_stp: 'USD',
+        },
+    };
+
+    let mock_props;
+
+    beforeEach(() => {
+        mock_props = {
+            is_logged_in: true,
+            platform: 'mt5',
+            show_eu_related: 'false',
+            residence: 'idn',
+            is_eu: 'false',
+            is_uk: 'false',
+        };
+    });
+
+    it('should render the proper column headers properly', () => {
+        render(<CompareAccountsContent {...mock_props} landing_companies={all_landing_companies} />);
+        expect(screen.getByText('Synthetic')).toBeInTheDocument();
+        expect(screen.getByText('CFDs')).toBeInTheDocument();
+        expect(screen.getByText('Financial STP')).toBeInTheDocument();
+    });
+
+    it('should render the proper data', () => {
+        render(<CompareAccountsContent {...mock_props} landing_companies={all_landing_companies} />);
+        expect(screen.getByText(/Account Currency/i)).toBeInTheDocument();
+        expect(screen.getByText('EUR')).toBeInTheDocument();
+        expect(screen.getByText('EUR/GBP/USD')).toBeInTheDocument();
+        expect(screen.getByText('USD')).toBeInTheDocument();
+
+        expect(screen.getAllByText(/Maximum leverage/i)[0]).toBeInTheDocument();
+        expect(screen.getAllByText(/1/i)[0]).toBeInTheDocument();
+        expect(screen.getByText(/Up to 1:1000/i)).toBeInTheDocument();
+        expect(screen.getByText(/Up to 1:30/i)).toBeInTheDocument();
+        expect(screen.getAllByText(/Up to 1:100/i)[0]).toBeInTheDocument();
+
+        expect(screen.getAllByText(/Order execution/i)[0]).toBeInTheDocument();
+        expect(screen.getAllByText(/2/i)[0]).toBeInTheDocument();
+        expect(screen.getAllByText(/Market/i)[0]).toBeInTheDocument();
+        expect(screen.getAllByText(/Market/i)[1]).toBeInTheDocument();
+        expect(screen.getAllByText(/Market/i)[2]).toBeInTheDocument();
+
+        expect(screen.getAllByText(/Spread/i)[0]).toBeInTheDocument();
+        expect(screen.getAllByText('Fixed/Variable')[0]).toBeInTheDocument();
+        expect(screen.getAllByText('Fixed/Variable')[1]).toBeInTheDocument();
+        expect(screen.getAllByText('Variable')[0]).toBeInTheDocument();
+
+        expect(screen.getAllByText(/Commission/i)[0]).toBeInTheDocument();
+        expect(screen.getAllByText(/4/i)[0]).toBeInTheDocument();
+        expect(screen.getAllByText(/No/i)[1]).toBeInTheDocument();
```

Comment:
If you want to test the text `No`, then may be u can directly test the length `expect(screen.getAllByText(/No/i).length).toBe(<expected count>)`

Reply:
Alrightss